### PR TITLE
Correctly building dummy block

### LINKED ISSUE #2160: grc:does not correcly  builds dummy blocks
If a flowgraph contains a block without a block description ( perhaps depreceated in 3.7 and now missing , or missing from other reasons ) grc should build a dummy block. https://github.com/gnuradio/gnuradio/blob/b1e0b54d02bb981eba734ea29174e6134950dde6/grc/core/FlowGraph.py#L371

But the dummy block will not be build:

    Loading: "/home/schroer/gnuradiocomponents/gr3.8tests/OptionalPadTest.grc"
    Block id "hier2Msg" not found
    Error: 'id'
>>> Failure
Traceback (most recent call last):
  File "/usr/local/gnuradio/lib/python3.6/dist-packages/gnuradio/grc/gui/MainWindow.py", line 261, in new_page
    file_path=file_path,
  File "/usr/local/gnuradio/lib/python3.6/dist-packages/gnuradio/grc/gui/Notebook.py", line 107, in __init__
    flow_graph.import_data(initial_state)
  File "/usr/local/gnuradio/lib/python3.6/dist-packages/gnuradio/grc/core/FlowGraph.py", line 377, in import_data
    block.import_data(**block_data)
  File "/usr/local/gnuradio/lib/python3.6/dist-packages/gnuradio/grc/core/blocks/block.py", line 344, in import_data
    self.params['id'].value = name
KeyError: 'id'

and in consequence, the flowgraph will not be build.
The fowgraph isn't   build even if the missing block is disabled.
So you have no chance to edit this  flowgraph and make it work again.

The only way to get this flowgraph building is to modify the grc file by hand.

### PR DESCRIPTION
**Pull request by dl1ksv**
This is a rebased version of #2197 and replaces #2197 directly.
As I failed to rebase directly I made this new pr. This should be more readable than the old one.
It fixes #2160.

### CODE REVIEW

**Comment by drmpeg** [comment]
I'm very much in favor of this PR. Since there's no version of gr-osmosdr for 3.8, almost all of the DTV example flow graphs fail with the cryptic "id not found" error. With this patch, they all work.

I was ready to re-generate all the DTV examples by hand, but this saves me from all that work.

**Comment by dl1ksv** [comment]
Is there a reason not merge this pr ? Including it's predecessor it's pending since November 2018.
It would make the transistion from 3.7 to 3.8 much easier. See [comment](https://github.com/gnuradio/gnuradio/pull/2278?_pjax=%23js-repo-pjax-container#issuecomment-451349695)

File: grc/core/FlowGraph.py [hunk 1]
```diff
@@ -423,6 +420,14 @@ def verify_and_get_port(key, block, dir):
                     src_blk_id, src_port_id, snk_blk_id, snk_port_id, e))
             had_connect_errors = True
 
+        for block in self.blocks:
+            if block.is_dummy_block :
+                block.rewrite()      # Make ports visible
+                # Flowgraph errors depending on disabled blocks are not displayed
+                # in ther error dialog box
```
Comment (dkozel) [inline grc/core/FlowGraph.py]: Typo, probably  "the"

**Comment by dkozel** [comment]
Merged manually so I could fix the typo. Thanks @dl1ksv, sorry for the really long delay here! You're absolutely right that this is going to be really helpful.

**Comment by nickoe** [comment]
Thanks you all

**Pull request closed without merging**